File: src/main/java/club/redux/sunset/lavafishing/client/renderer/BlockEntity/BlockEntityRendererPrometheusBounty.java
package club.redux.sunset.lavafishing.client.renderer.BlockEntity;

import club.redux.sunset.lavafishing.block.BlockEntity.BlockEntityPrometheusBounty;
import club.redux.sunset.lavafishing.util.Reference;
import net.minecraft.MethodsReturnNonnullByDefault;
import net.minecraft.client.renderer.Sheets;
import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider;
import net.minecraft.client.renderer.blockentity.ChestRenderer;
import net.minecraft.client.resources.model.Material;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.level.block.state.properties.ChestType;

import javax.annotation.ParametersAreNonnullByDefault;

@MethodsReturnNonnullByDefault
@ParametersAreNonnullByDefault
public class BlockEntityRendererPrometheusBounty extends ChestRenderer<BlockEntityPrometheusBounty> {
    public static BlockEntityRendererPrometheusBounty instance;

    public BlockEntityRendererPrometheusBounty(BlockEntityRendererProvider.Context context) {
        super(context);
        instance = this;
    }

    @Override
    protected Material getMaterial(BlockEntityPrometheusBounty blockEntity, ChestType chestType) {
        return new Material(Sheets.CHEST_SHEET, new ResourceLocation(Reference.MOD_ID, "entity/chest/prometheus_bounty"));
    }
}

File: src/main/kotlin/club/redux/sunset/lavafishing/client/renderer/blockentity/BlockEntityRendererPrometheusBounty.kt
package club.redux.sunset.lavafishing.client.renderer.blockentity

import club.redux.sunset.lavafishing.LavaFishing
import club.redux.sunset.lavafishing.block.blockentity.BlockEntityPrometheusBounty
import club.redux.sunset.lavafishing.registry.ModBlockEntityTypes
import net.minecraft.client.renderer.Sheets
import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider
import net.minecraft.client.renderer.blockentity.ChestRenderer
import net.minecraft.client.renderer.blockentity.state.ChestRenderState
import net.minecraft.client.resources.model.sprite.SpriteId
import net.neoforged.neoforge.client.event.EntityRenderersEvent

class BlockEntityRendererPrometheusBounty(
    context: BlockEntityRendererProvider.Context,
) : ChestRenderer<BlockEntityPrometheusBounty>(context) {
    override fun getCustomSprite(blockEntity: BlockEntityPrometheusBounty, state: ChestRenderState) =
        SpriteId(Sheets.CHEST_SHEET, LavaFishing.identifier("entity/chest/prometheus_bounty"))

    companion object {
        fun onRegisterRenderers(event: EntityRenderersEvent.RegisterRenderers) {
            event.registerBlockEntityRenderer(
                ModBlockEntityTypes.PROMETHEUS_BOUNTY.get(),
                ::BlockEntityRendererPrometheusBounty
            )
        }
    }
}
